it/CreateChildLinkFromLegalUnitSpec.scala

```diff
@@ -0,0 +1,166 @@
+
+  private val RegisterPeriod = Period.fromYearMonth(2018, MARCH)
+  private val VatUnitAcronym = toAcronym(ValueAddedTax)
+  private val VatRef = "401347263289"
+  private val VatUnitId = UnitId(VatRef)
+  private val LegalUnitAcronym = toAcronym(LegalUnit)
+  private val TargetUBRN = "1000012345000999"
+  private val TargetLegalUnitId = UnitId(TargetUBRN)
+  private val Family = HBaseRestUnitLinksRepository.ColumnFamily
+  private val EnterpriseAcronym = toAcronym(Enterprise)
+
+  private val HBaseCreateColumnRequestBody =
+    s"""{"Row": ${
+      List(
+        aRowWith(key = s"${UnitLinksRowKey(TargetLegalUnitId, LegalUnit)}", columns =
+          aColumnWith(Family, qualifier = UnitLinksQualifier.toChild(VatUnitId), value = VatUnitAcronym, timestamp = None))
+      ).mkString("[", ",", "]")
+    }}"""
+
+  private val UnitLinksForLegalUnitHBaseResponseBody =
+    s"""{ "Row": ${
+      List(
+        aRowWith(key = s"${UnitLinksRowKey(TargetLegalUnitId, LegalUnit)}", columns =
+          aColumnWith(Family, qualifier = UnitLinksQualifier.toParent(Enterprise), value = "1234567890"),
+          aColumnWith(Family, qualifier = UnitLinksQualifier.toChild(UnitId("87654321")), value = toAcronym(CompaniesHouse)))
+      ).mkString("[", ",", "]")
+    }}"""
+
+  info("As a business register subject matter expert")
+  info("I want to move an incorrectly linked VAT to a different Legal Unit")
+  info("So that I can improve the quality of the business register")
+
+  feature("create a link to a child VAT unit from a Legal Unit") {
+    scenario("when then Legal Unit exists in the register and the VAT unit exists in the admin data") { wsClient =>
+      Given(s"there exists a Legal Unit with UBRN $TargetUBRN")
+      stubHBaseFor(aUnitLinksExactRowKeyRequest(withUnitId = TargetLegalUnitId, withUnitType = LegalUnit, withPeriod = RegisterPeriod).
+        willReturn(anOkResponse().withBody(UnitLinksForLegalUnitHBaseResponseBody)))
+      And(s"the admin data does contain a VAT unit with reference $VatRef")
+      stubAdminDataApiFor(aVatRefLookupRequest(VatUnitId, RegisterPeriod).willReturn(anOkResponse()))
+      And(s"a database request to create a child link from $TargetUBRN to the VAT unit with reference $VatRef will succeed")
+      stubHBaseFor(aCreateUnitLinkRequest(withUnitType = LegalUnit, withUnitId = TargetLegalUnitId, withPeriod = RegisterPeriod).
+        withRequestBody(equalToJson(HBaseCreateColumnRequestBody)).
+        willReturn(anOkResponse()))
+
+      When(s"the creation of a child link from $TargetUBRN to the VAT unit with reference $VatRef is requested")
+      val response = await(wsClient.url(s"/v1/periods/${Period.asString(RegisterPeriod)}/types/$LegalUnitAcronym/units/$TargetUBRN").
+        withHeaders(CONTENT_TYPE -> JsonPatchMediaType).
+        patch(s"""[{"op": "add", "path": "/children/$VatRef", "value": "$VatUnitAcronym"}]"""))
+
+      Then(s"a Success response is returned")
```

Comment:
The idea here was to try and focus the language of this scenario on "success" rather than "no content".  

No Content is the status code we have chosen to indicate success for this operation, and our assertion obviously tests the specific value.  However, a more casual reader just looking at the Given / When / Then should be able to understand the purpose of the test without being overly concerned with 204 vs 200 etc. - they just care that this "happy path" scenario returns some indicator of success...

Having said that, the sad path scenarios inevitably get much more specific, and so map to explicit status codes.  So I appreciate that there is a discrepancy of approach here.  Happy to change if you feel strongly about it ...